dataprocspawner/spawner.py

```diff
@@ -1452,8 +1441,11 @@ def _build_cluster_config(self, cluster_data=None):
     if self.force_add_jupyter_component:
       if Component['JUPYTER'].value not in optional_components:
         optional_components.append(Component['JUPYTER'].value)
-      if Component['ANACONDA'].value not in optional_components:
-        optional_components.append(Component['ANACONDA'].value)
+      # preview (2.x) images and up do not support Anaconda
+      if self._image_version_supports_anaconda(
```

Comment:
Should we raise an exception somewhere if the provided image does not support Anaconda?

Reply:
No, the image still supports Jupyter. We just use Miniconda instead of Anaconda now, so you can't use the Anaconda component.

I will actually change this to remove ANACONDA from the list if it's present on more recent images.